api: add handle to/from int conversion APIs (#7245)
pmodels/mpich: src/binding/c/io_api.txt

#endif
}

MPI_File_toint:
    .return: INTEGER
    file: FILE, [file handle]

>     .desc: Translates a C file handle to a Fortran file handle

Reviewer: ```suggestion
    .desc: converts MPI_File to an integer handle
```

Author: fixed

    .impl: direct
{
#ifndef HAVE_ROMIO

>     return 0;

Reviewer: See this comment of mine https://github.com/mpi-forum/header_and_stub_library/pull/39#issuecomment-2647169296 
and @jeffhammond's reply https://github.com/mpi-forum/header_and_stub_library/pull/39#issuecomment-2656390107

>     return 0;

Reviewer 2: OK so basically you can't create a valid file handle without ROMIO so a user shouldn't need to convert anything. Would it be better to return `(int)MPI_FILE_NULL` since it is non-zero in the ABI spec?

>     return 0;

Reviewer: Alternatively, maybe it is better to keep returning `0`, or even `-1` , that would provide a programmatic way to check for MPICH built without ROMIO. That is, `MPI_File_toint(MPI_FILE_NULL) == 0` means ROMIO is not available.

>     return 0;

Author: That brings out an issue: ROMIO's `MPI_File_c2f` returns integer from 1 and up; thus it *requires* `MPI_File_c2f(MPI_FILE_NULL)` return `0`.  This works when MPICH's `MPI_FILE_NULL` is `NULL` pointer.

@raffenet Without the understanding that MPICH's `MPI_FILE_NULL` is `(void *) 0`, I think `(int) MPI_FILE_NULL` is quite misleading. It is as if we can directly cast pointer to integer. I would prefer `return 0;`.

NOTE: I need add a commit to fix `ABI_File_{to,from}_mpi` so it special treat the case of `MPI_FILE_NULL`.

>     return 0;

Reviewer 2: > @raffenet Without the understanding that MPICH's `MPI_FILE_NULL` is `(void *) 0`, I think `(int) MPI_FILE_NULL` is quite misleading. It is as if we can directly cast pointer to integer. I would prefer `return 0;`.

Are you saying it is misleading to people reading the source code? Or just that we are returning a handle value when the input was likely garbage? I think I'm fine either way using 0 since we are in a scenario where the users shouldn't be calling these functions, just trying to understand the discussion.

#else
    return (int) MPIR_File_c2f_impl(file);
#endif
}

MPI_File_fromint:
    .return: FILE
    file: INTEGER

>     .desc: Translates a Fortran file handle to a C file handle

Reviewer: ```suggestion
    .desc: converts an integer handle to MPI_File
```

Author: fixed.
